libs/ballot-interpreter-nh/src/convert.ts

```diff
@@ -168,72 +217,198 @@ export function readGridFromElectionDefinition(
   );
 }
 
+/**
+ * Kinds of errors that can occur when converting a ballot card definition.
+ */
+export enum ConvertErrorKind {
+  ElectionValidationFailed = 'ElectionValidationFailed',
+  InvalidBallotSize = 'InvalidBallotSize',
+  InvalidDistrictId = 'InvalidDistrictId',
+  InvalidElectionDate = 'InvalidElectionDate',
+  InvalidTimingMarkMetadata = 'InvalidTimingMarkMetadata',
+  MismatchedBallotImageSize = 'MismatchedBallotImageSize',
+  MismatchedOvalGrids = 'MismatchedOvalGrids',
+  MissingBallotSize = 'MissingBallotSize',
+  MissingCandidateName = 'MissingCandidateName',
+  MissingElectionDate = 'MissingElectionDate',
+  MissingElectionId = 'MissingElectionId',
+  MissingElectionName = 'MissingElectionName',
+  MissingOfficeName = 'MissingOfficeName',
+  MissingPrecinctId = 'MissingPrecinctId',
+  MissingTimingMarkMetadata = 'MissingTimingMarkMetadata',
+  MissingTownId = 'MissingTownId',
+  MissingTownName = 'MissingTownName',
```

Comment:
wondering if we'll find this level of granularity in types useful, or onerous in the long run.

consider maybe `MissingRequiredData` and then a detailed error message that says what's missing.

Not a blocker, ok with the current approach, only a suggestion for consideration.

Reply:
Yep, that's a good point. I consolidated most of those down to one error type.